src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -549,6 +550,10 @@ public RubyNode visitCallNode(CallParseNode node) {
                 final RubyNode ret = AssertNotCompiledNodeGen.create();
                 ret.unsafeSetSourceSection(sourceSection);
                 return addNewlineIfNeeded(node, ret);
+            } else if (methodName.equals("assert_never_part_of_compilation")) {
+                final RubyNode ret = AssertNeverPartOfCompilationNodeGen.create();
+                ret.unsafeSetSourceSection(sourceSection);
+                return addNewlineIfNeeded(node, ret);
```

Comment:
Why not make it a TrufflePrimtiive if it's meant to be AST-inlined?
I think that would be cleaner than adding another special treatment in the translator.

Reply:
I kept it consistent with the other Graal asserts.

Reply:
Also, when using a debugging tool like this I think you don't want to be relying on inlining (even if it's forced) to work correctly. Primitives are still inlined aren't they?

Reply:
Right, those should probably be primitives too (but out of scope of this PR), it would avoid all these special cases and the need to define a fallback method that just raises.

Reply:
> Also, when using a debugging tool like this I think you don't want to be relying on inlining (even if it's forced) to work correctly. Primitives are still inlined aren't they?

Primitive don't use force inlining, they just embed the node directly in the AST (same CallTarget), just like this does, but in a generic way.

Reply:
FYI @pitr-ch moved these to `TrufflePrimitive` in fcc4403c123abe6b91127b181f6b4af33cb16808.